src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -106,18 +107,33 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Diagno
             bool hasErrors,
             DiagnosticBag diagnostics)
         {
+            return BindPattern(node, inputType, inputValEscape, out _, out _, permitDesignations, hasErrors, diagnostics);
+        }
+
+        internal BoundPattern BindPattern(
+            PatternSyntax node,
+            TypeSymbol inputType,
+            uint inputValEscape,
+            out TypeSymbol outputType,
```

Comment:
wouldn't it be simpler to just add this to BoundPattern?

Reply:
Possible, but for most patterns it is trivial to compute locally.

---
In reply to: [388722825](https://github.com/dotnet/roslyn/pull/42207#discussion_r388722825) [](ancestors = 388722825)